src/storage/DataAccessorBasedStore.ts

```diff
@@ -347,8 +348,23 @@ export class DataAccessorBasedStore implements ResourceStore {
    * @param slug - Slug to use for the new URI.
    */
   protected createURI(container: ResourceIdentifier, isContainer: boolean, slug?: string): ResourceIdentifier {
-    return { path:
-        `${ensureTrailingSlash(container.path)}${slug ? trimTrailingSlashes(slug) : uuid()}${isContainer ? '/' : ''}` };
+    const base = ensureTrailingSlash(container.path);
+    const cleanedSlug = slug ? this.cleanSlug(slug) : null;
+    const name = cleanedSlug ?? uuid();
```

Comment:
The first line reassigns slug to `null` if it is `undefined`. Is there a reason you did it like this and not the line Ruben suggested? In case it was because of the linter, the way to write it so the linter is happy is `(slug && this.cleanSlug(slug)) ?? uuid();`